torch/_functorch/aot_autograd.py

```diff
@@ -2114,8 +2134,11 @@ def create_aot_dispatcher_function(
 
         def process_inputs(flat_args):
             if config.use_fake_tensor or isinstance(fake_mode, FakeTensorMode):
-
                 def convert(idx, x):
+                    if config.use_dynamic_shapes:
+                        from torch._dynamo.source import ConstantSource
+                        if isinstance(x, int):
+                            return shape_env.create_symintnode(shape_env.create_symbol(x, ConstantSource(f"sym_{idx}")))
```

Comment:
The ConstantSource here is wrong (and this matters if we actually want to perform a guard on the passed in int.) If this happened in forwards, the correct layering is to SymInt'ify the int input at the Dynamo layer. However, this is happening in backwards, which means that actually the Source here has to hook into some as yet unimplemented second level guard mechanism for the backwards function.

Reply:
Yes, we discussed this on VC last night - ConstantSource is for something else. Let me know if you need a hand figuring out the correct source.